src/test/java/seedu/address/model/volunteer/VolunteerTest.java

```diff
@@ -0,0 +1,100 @@
+package seedu.address.model.volunteer;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static seedu.address.testutil.Assert.assertThrows;
+import static seedu.address.testutil.TypicalVolunteers.ALICE;
+import static seedu.address.testutil.TypicalVolunteers.BOB;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.address.testutil.VolunteerBuilder;
+
+public class VolunteerTest {
+
+    @Test
+    public void asObservableList_modifyList_throwsUnsupportedOperationException() {
+        Volunteer volunteer = new VolunteerBuilder().build();
+        if (volunteer.getInvolvedIn().isEmpty()) {
+            assertTrue(volunteer.getInvolvedIn().isEmpty());
+        } else {
+            assertThrows(UnsupportedOperationException.class, () -> volunteer.getEvents().remove(0));
+        }
+    }
+
+    @Test
+    public void isSameVolunteer() {
+        // same object -> returns true
+        assertTrue(ALICE.isSameVolunteer(ALICE));
+
+        // null -> returns false
+        assertFalse(ALICE.isSameVolunteer(null));
+
+        // same name, all other attributes different -> returns true
+        Volunteer editedAlice = new VolunteerBuilder(ALICE).withPhone("87654321").withEmail("alice@example.com")
+                .withAvailableDate("2023-12-12").withStartTimeAvailability("07:00").withEndTimeAvailability("17:00")
+                .build();
+        assertTrue(ALICE.isSameVolunteer(editedAlice));
+
+        // different name, all other attributes same -> returns false
+        Volunteer editedAliceWithDifferentName = new VolunteerBuilder(ALICE).withName("Alice Ong").build();
+        assertFalse(ALICE.isSameVolunteer(editedAliceWithDifferentName));
+
+        // name differs in case, all other attributes same -> returns false
+        Volunteer editedBob = new VolunteerBuilder(BOB).withName("bob tan").build();
+        assertFalse(BOB.isSameVolunteer(editedBob));
+
+        // name has trailing spaces, all other attributes same -> returns false
+        Volunteer editedBobWithTrailingSpaces = new VolunteerBuilder(BOB).withName("Bob Tan ").build();
```

Comment:
If we have time maybe we can try to look into how to prevent `Bob Tan` from being a different volunteer when compared to `Bob Tan ` but I think its fine even if we leave it as is for now.